dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ResourcesServiceImpl.java

```diff
@@ -318,6 +340,128 @@ public Result<Object> createResource(User loginUser,
         return result;
     }
 
+    /**
+     * create batch resource
+     *
+     * @param loginUser  login user
+     * @param files      files
+     * @param type       type
+     * @param pid        parent id
+     * @param currentDir current directory
+     * @return create result code
+     */
+    @Override
+    @Transactional
+    public Result<Object> createBatchResources(User loginUser,
+                                               ResourceType type,
+                                               List<MultipartFile> files,
+                                               int pid,
+                                               String currentDir) {
+        Result<Object> result = new Result<>();
+        String funcPermissionKey = type.equals(ResourceType.FILE) ? ApiFuncIdentificationConstant.FILE_UPLOAD
+                : ApiFuncIdentificationConstant.UDF_UPLOAD;
+        boolean canOperatorPermissions =
+                canOperatorPermissions(loginUser, null, AuthorizationType.RESOURCE_FILE_ID, funcPermissionKey);
+        if (!canOperatorPermissions) {
+            putMsg(result, Status.NO_CURRENT_OPERATING_PERMISSION);
+            return result;
+        }
+        result = checkResourceUploadStartupState();
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+        result = verifyPid(loginUser, pid);
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+
+        // make sure login user has tenant
+        String tenantCode = getTenantCode(loginUser.getId(), result);
+        if (StringUtils.isEmpty(tenantCode)) {
+            return result;
+        }
+
+        result = verifyFiles(type, files);
+        if (!result.getCode().equals(Status.SUCCESS.getCode())) {
+            return result;
+        }
+        Map<String, MultipartFile> fileMap = files.stream().collect(Collectors.toMap(key -> key.getOriginalFilename(), value -> value, (isOld, isNew) -> isOld));
+
+        Object[] filesNameSet = fileMap.keySet().toArray();
```

Comment:
Please change to `String[] filesNameSet = fileMap.keySet().toArray(new String[0]);`